lib/trino-filesystem/src/main/java/io/trino/filesystem/fileio/ForwardingFileIo.java

```diff
@@ -61,4 +66,22 @@ public void deleteFile(String path)
             throw new UncheckedIOException("Failed to delete file: " + path, e);
         }
     }
+
+    @Override
+    public void deleteFiles(Iterable<String> pathsToDelete)
+            throws BulkDeletionFailureException
+    {
+        Iterable<List<String>> partitions = Iterables.partition(pathsToDelete, DELETE_BATCH_SIZE);
+        partitions.forEach(this::deleteBatch);
+    }
+
+    private void deleteBatch(List<String> filesToDelete)
+    {
+        try {
+            fileSystem.deleteFiles(filesToDelete);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Failed to delete some or all of files: " + filesToDelete, e);
```

Comment:
I don't think including all file paths to the error message is a good idea. It may include 1,000 files in large deletes.

Reply:
Hi  @ebyhr| @findepi | @electrum , I am proposing a PR https://github.com/trinodb/trino/pull/16006 to track the number of files that failed to delete. If https://github.com/trinodb/trino/pull/16006 gets approved then we can throw `BulkDeletionFailureException` over here.

Reply:
@ebyhr you convinced me, let's not make the message that long. OTOH, it could be problematic if we do not include any paths in the message. It can make identifying problems harder (eg which bucket did we try to access?).
I pushed a change to display only first 5 paths and then ellipsis `...`.